tests/test_beamstrahlung.py

```diff
@@ -0,0 +1,327 @@
+        # only if BS on
+        slices_other_beam_zeta_bin_width = np.abs(np.diff(slicer.bin_edges))
+        
+        # change nans to 0
+        slices_other_beam_x_center[np.isnan(slices_other_beam_x_center)] = 0
+        slices_other_beam_zeta_center[np.isnan(slices_other_beam_zeta_center)] = 0
+        slices_other_beam_Sigma_11[np.isnan(slices_other_beam_Sigma_11)] = 0
+        slices_other_beam_Sigma_22[np.isnan(slices_other_beam_Sigma_22)] = 0
+        slices_other_beam_Sigma_33[np.isnan(slices_other_beam_Sigma_33)] = 0
+        slices_other_beam_Sigma_44[np.isnan(slices_other_beam_Sigma_44)] = 0
+            
+        el_beambeam_b1 = xf.BeamBeamBiGaussian3D(
+                _context=context,
+                config_for_update = config_for_update,
+                other_beam_q0=1,
+                phi=phi,
+                alpha=0,
+                # decide between round or elliptical kick formula
+                min_sigma_diff     = 1e-28,
+                # slice intensity [num. real particles] n_slices inferred from length of this
+                slices_other_beam_num_particles      = bunch_intensity/n_macroparticles_b2*slices_other_beam_num_particles,
+                # unboosted strong beam moments  
+                slices_other_beam_x_center    = slices_other_beam_x_center,
+                slices_other_beam_zeta_center = slices_other_beam_zeta_center,
+                slices_other_beam_Sigma_11    = slices_other_beam_Sigma_11,
+                slices_other_beam_Sigma_22    = slices_other_beam_Sigma_22,
+                slices_other_beam_Sigma_33    = slices_other_beam_Sigma_33,
+                slices_other_beam_Sigma_44    = slices_other_beam_Sigma_44,
+                # only if BS on
+                flag_beamstrahlung = 1,
+                slices_other_beam_zeta_bin_width_star = slices_other_beam_zeta_bin_width/np.cos(phi),  # boosted dz
+                # has to be set
+                slices_other_beam_Sigma_12    = n_slices*[0],
+                slices_other_beam_Sigma_34    = n_slices*[0],
+        )
+        el_beambeam_b1.name = "beambeam"
+        
+        #########################
+        # track for 1 collision #
+        #########################
+        
+        line = xt.Line(elements = [el_beambeam_b1])
+        
+        tracker = xt.Tracker(line=line)
+        record = tracker.start_internal_logging_for_elements_of_type(xf.BeamBeamBiGaussian3D, capacity={"beamstrahlungtable": int(3e5)})
+        tracker.track(particles_b1, num_turns=1)
+        tracker.stop_internal_logging_for_elements_of_type(xf.BeamBeamBiGaussian3D)
+
+        ###################################
+        # compare spectrum with guineapig #
+        ###################################
+
+        fname="../test_data_beamstrahlung/guineapig_ttbar2_beamstrahlung_photon_energies_gev.txt"
+        guinea_photons = np.loadtxt(fname)  # contains about 250k photons emitted from 1e6 macroparticles in 1 collision
```

Comment:
added beamstrahlung parameter check to the test. The test with the sigma_z blowup requires tracking with the arc so I did not add that. Also I did not add the check against the Guineapig spectrum to the examples because Guineapig simulates only a signle beambeam collision and in the examples I track and have a spectrum from many turns.